packages/cli/src/commands/generate.js

```diff
@@ -33,6 +33,11 @@ export default {
           options.modern = 'client'
         }
       }
+    },
+    'fail-on-page-error': {
+      type: 'boolean',
+      default: false,
+      description: 'Exit with non-zero status code if there are errors when generating pages'
```

Comment:
I am open for naming suggestions for this option.

Reply:
I think this can be simply `fail-on-error` to fail on any kind of error. It may happen on middleware, store or plugins too.

Reply:
Sorry, I missed this comment. Renamed.

Reply:
I think we should strive to always halt on errors, but as a non-zero exit code for page errors would be a breaking change on eg CI deployments we cant do that now until v3. But as said before, then even with v3 you still want to differentiate between page errors and other errors. Unless its ok that `--no-fail-on-error` only doesnt fail page errors. But might be confusing.